data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/sink/FileSink.java

```diff
@@ -77,14 +78,19 @@ public void output(final Collection<Record<Object>> records) {
                 return;
 
             for (final Record<Object> record : records) {
+                if (record.getData() instanceof Event) {
```

Comment:
This logic can go into `SinkForwardRecordsContext`.

```
public void addRecord(Record<?> record) {
  if(! (record.getData() instanceof Event)) {
    return;
  }
  ...
```

Reply:
That means we can't forward `Record<Object>` records. I was testing with File sink forwarding to another file sink to verify the forwarding logic.